### DISCUSSION TITLE
apollo-server-express inside app.use

### DISCUSSION THREAD

**Discussion by codelover2k** [apollographql/apollo-server#5872] [Q&A]
is it possible to put apollo-server-express middleware inside app.use, so you can execute a middleware before, I mean something like this:

```js
await apolloServer.start();

app.use('/api/graphql', myLogMiddleware, checkAuthentication, apolloServer.getMiddleware({
  // not needed
  // path: '/api/graphql',
  cors: myCors,
  graphiql: process.env.NODE_ENV === 'development',
}));
```

**Comment by glasser** [comment]
Yes, though I think you might have to pass `path: ''` or something? Hoping in AS4 to get rid of the weird `path` argument and just have you use the normal argument to `app.use` to specify the path.

**Reply by codelover2k** [reply]
hi glasser and thanks for response!

What works is of corse this:
```js
await apolloServer.start();

app.use(myLogMiddleware, checkAuthentication, apolloServer.getMiddleware({
  path: '/api/graphql',
  cors: myCors,
  graphiql: process.env.NODE_ENV === 'development',
}));
```

The problem is that with this code, every request that is not caught above, will land in this middleware.

What I also tried is this:
```js
app.use('/api/graphql', myLogMiddleware, checkAuthentication, apolloServer.getMiddleware({
  path: '/api/graphql',
  cors: myCors,
  graphiql: process.env.NODE_ENV === 'development',
}));
```
This does not work and I do not understand why, I mean I want to use the path to explicitly tell express when it can execute the middleware handlers and when to skip them.  The `/api/graphql` route in this example is skipped entirely.

Would be nice if there is a workaround :)

**Reply by glasser** [reply]
Does it work with `path: ''`?

**Reply by codelover2k** [reply]
if you look at this minimal example:
```js
app.use((req, res, next) => {
  console.log("I AM GRAPHQL!");
  next();
  }, 
  apolloServer.getMiddleware({
    path: '/api/graphql',
    cors: myCors,
    graphiql: process.env.NODE_ENV === 'development',
}));
```
The url `/api/graqhql` works and the middleware before is triggered.
But if you use any other url that is not caught before, it will trigger the middleware before `apolloGraphQLServer` only, and then you will get a 404, because `/api/graphqlx` does not exists.

**Comment by codelover2k** [comment]
anything new about this topic?

**Reply by glasser** [reply]
from a high level, we will soon be announcing an Apollo Server 4 roadmap that will include entirely removing path sensitivity from AS as mentioned in my previous response.
